lib/CodeGen/CGCall.cpp

```diff
@@ -175,9 +175,25 @@ arrangeLLVMFunctionInfo(CodeGenTypes &CGT, bool instanceMethod,
   appendParameterTypes(CGT, prefix, paramInfos, FTP);
   CanQualType resultType = FTP->getReturnType().getUnqualifiedType();
 
+  // HCC (HIP) specific: for HIP we want __global__ functions to represent
+  // actual kernel entry-points, i.e. we want them to have AMDGPU_KERNEL as
+  // their calling convention. The only way to get that early enough appears to
+  // be the below, wherein the mark them as OpenCL kernels before the layout is
+  // generated. This is temporary and somewhat unpleasant, the correct solution
+  // would be to bubble up AMDGPU_KERNEL as a full blown calling convention
+  // which can be used orthogonally to OpenCL, such as e.g. __stdcall.
+  FunctionType::ExtInfo Tmp = FTP->getExtInfo();
```

Comment:
Usually they don't use name like Tmp but use name like Info or ExtInfo. Otherwise LGTM.

Reply:
@yxsamliu good observation, thank you. I would like to not try to upstream this tough, unless we really need it, since it is not really nice, as opposed to adding the calling convention properly. Would it be acceptable to look upon this as a temporary band-aid?

Reply:
OK. LGTM.